common/windows/win32_window.cc

```diff
@@ -0,0 +1,150 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "win32_window.h"
+#include "shellscalingapi.h"
+
+// the Windows DPI system is based on this
+// constant for machines running at 100% scaling.
+constexpr int base_dpi = 96;
+
+// Scale helper to convert logical scaler values to physical using passed in
+// scale factor
+int Scale(int source, double scale_factor) {
+  return static_cast<int>(static_cast<double>(source * scale_factor));
+}
+
+Win32Window::Win32Window() {}
+
+Win32Window::~Win32Window() { Destroy(); }
+
+bool Win32Window::CreateAndShow(const char *title, const unsigned int x,
+                                const unsigned int y, const unsigned int width,
+                                const unsigned int height) {
+  Destroy();
+
+  WNDCLASS window_class = ResgisterWindowClass(title);
+
+  // New windows are created on the default monitor.  Window sizes are specified
+  // to the OS in physical pixels, hence to ensure a consistent size to will
+  // treat the width height passed in to this function as logical pixels and
+  // scale to appropriate for the default monitor.
+  HMONITOR defaut_mon = MonitorFromWindow(nullptr, MONITOR_DEFAULTTOPRIMARY);
+  UINT dpi_x = 0, dpi_y = 0;
+  GetDpiForMonitor(defaut_mon, MDT_EFFECTIVE_DPI, &dpi_x, &dpi_y);
+
+  double scaleFactor = static_cast<double>(dpi_x) / base_dpi;
+
+  auto window = CreateWindow(
```

Comment:
I don't know offhand what the type of this is, so I would prefer not using `auto`